thrifty-runtime/src/main/java/com/microsoft/thrifty/transport/URLTransport.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Thrifty
+ *
+ * Copyright (c) Microsoft Corporation
+ *
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING
+ * WITHOUT LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE,
+ * FITNESS FOR A PARTICULAR PURPOSE, MERCHANTABLITY OR NON-INFRINGEMENT.
+ *
+ * See the Apache Version 2.0 License for specific language governing permissions and limitations under the License.
+ */
+package com.microsoft.thrifty.transport;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+public class URLTransport extends Transport {
+
+    private URL url;
+    private int timeout = 0;
+
+    private HttpURLConnection connection;
+    private InputStream inputStream;
+    private ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+
+    public static class Builder {
+
+        private URL url;
+        private int timeout = 0;
+
+        public Builder(URL url) {
+            this.url = url;
+        }
+
+        public Builder setTimeout(int timeout) {
+            if (timeout < 0) {
+                throw new IllegalArgumentException("timeout can not be negative");
+            }
+            this.timeout = timeout;
+            return this;
+        }
+
+        public URLTransport build() {
+            return new URLTransport(this);
+        }
+    }
+
+    URLTransport(Builder builder) {
+        this.url = builder.url;
+        this.timeout = builder.timeout;
+    }
+
+    @Override
+    public int read(byte[] buffer, int offset, int count) throws IOException {
+        return inputStream.read(buffer, offset, count);
+    }
+
+    @Override
+    public void write(byte[] buffer, int offset, int count) throws IOException {
+        outputStream.write(buffer, offset, count);
+    }
+
+    @Override
+    public void flush() throws IOException {
+
+        HttpURLConnection connection = (HttpURLConnection) url.openConnection();
```

Comment:
I think this logic is in the wrong place.  Evidently, this is connection-establishment and initialization code, which definitely shouldn't be in `flush()`, which can be called any number of times by protocol classes.  Shouldn't this be part of the builder?  Or, at the very least, in an `open()` method of its own?

More worryingly, `inputStream` is being initialized in `flush()`, which is definitely wrong.  That implies that `read()` will always fail before `flush()`.  There's no guarantee that a protocol will `flush()` its transport, so this method is certainly not the place to be preparing for reads.